test/yaml-rest-runner/src/main/java/org/elasticsearch/test/rest/yaml/section/SkipSection.java

```diff
@@ -88,34 +115,43 @@ void validate(XContentLocation contentLocation) {
                 throw new ParsingException(contentLocation, "reason is mandatory within skip version section");
             }
             // make feature "skip_os" mandatory if os is given, this is a temporary solution until language client tests know about os
-            if (operatingSystems.isEmpty() == false && testFeatures.contains("skip_os") == false) {
+            if (operatingSystems.isEmpty() == false && requiredYamlRunnerFeatures.contains("skip_os") == false) {
                 throw new ParsingException(contentLocation, "if os is specified, feature skip_os must be set");
             }
-            if (xpackRequested == XPackRequested.MISMATCHED) {
+            if (xpackRequired == XPackRequired.MISMATCHED) {
                 throw new ParsingException(contentLocation, "either `xpack` or `no_xpack` can be present, not both");
             }
+            if (forbiddenClusterFeatures.stream().anyMatch(x -> requiredClusterFeatures.contains(x))) {
```

Comment:
nit - can use `Sets.haveNonEmptyIntersection(forbidden, required)`

Reply:
Thank you! I was looking for non-mutable intersection functions and could not find them. Will definitely replace